blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+
+We use CI/CD pipelines and automatizations in every project we are responsible for. Especially Github Actions is becoming more popular, as it’s simple to use and growing in functionality. Over the last couple of months, we managed to create multiple projects and repositories that are using Github Actions extensively.
+
+Github Actions provides a generous free tier, so costs for smaller projects are often nonexistent. Once you reach the free tier limits, every Action (and other services) needs to be "paid by use" at the end of the month. Github aggregates all costs and shows them in your settings. But Github doesn't provide charts or a dashboard to get into the details of the costs, for example for historical data.
+
+This can be a problem if you have multiple projects, as it’s not obvious which service causes the costs. If you want to figure that out, you can get a CSV from Github, which is hard to read and analyze. We, therefore, decided to create a dashboard to visualize this CSV and show you the costs per repository and month.
+
+![screenshot of the billing settings in a Github account](./images/screenshot-of-gh-billing-settings.png)
+
+This is all the billing information you get from the Github Settings. No historical data, no expense per service.
+
+![screenshot of the Github billing dashboard with some charts](./images/screenshot-of-gh-dashboard.png)
+
+And this is what you get from the Github Billing Dashboard. Costs per service per month with historic data.
+
+This blog post gives you some insights into how we build it.
+
+## Technologies and Libraries
+
+The dashboard runs locally in the user's browser and thus does not send any data to external servers, even the most recently used files are stored only in the browser's local storage. The dashboard was mainly developed with React and Typescript. For the styling of the dashboard, we used styled-components. In addition, various libraries were used.
+
+### papaparse
+
+[papaparse](https://github.com/mholt/PapaParse) is a CSV parser that converts CSV to JSON. Besides strings, papaparse can process CSV files directly, no matter if they are located locally on the computer or have to be accessed via an external URL.
+
+The following example shows how easy the conversion of a CSV file is. The function parse() is called with a file like sample.csv. Additionally, it needs as a second input parameter a config object, in which certain options for the parsing can be set. For example, the headings of the individual values are camalized here. The parsing itself, the assignment of the values to the headings, and the transformation into the UsageReportCsvEntry type are completely handled by papaparse. The parsed object can still be modified. In the sample code, you can see how in addition to the existing CSV file headers "pricePerUnit" is calculated and added to the finished object.
```

Comment:
```suggestion
The following example shows how easy the conversion of a CSV file is. The function parse() is called with a file like sample.csv. Additionally, it needs as a second input parameter a config object, in which certain options for the parsing can be set. For example, the headings of the individual values are camalized here. The parsing itself, the assignment of the values to the headings, and the transformation into the `UsageReportCsvEntry` type are completely handled by papaparse. The parsed object can still be modified. In the sample code, you can see how in addition to the existing CSV file headers `pricePerUnit` is calculated and added to the finished object.
```

Reply:
backtick makes here sense, doesn't it ?

Reply:
> One other thing: Can you search for some other picture? The current one is from the example blog post.
> The keyword dashboard brings so many ideas. It only needs to be symbolic. Or we can ask verena for some photo collage where a stock photo is merged with a nice screenshot of the tool's output ?
> 
> * https://unsplash.com/photos/4VSsq9ErzOs
> * https://unsplash.com/photos/hpjSkU2UYSU (for a collage)
> * etc etc.
> 
> Don't forget the attribution data for photos in the frontmatter header though 🙏

I like the picture with the cockpit because it is definitely causally related to the project👌  We have additionally asked Verena, maybe she also has an idea. Until then I would use the cockpit image.